digiwf-engine/digiwf-engine-service/src/test/java/de/muenchen/oss/digiwf/engine/incidents/IncidentNotifierHandlerTest.java

```diff
@@ -0,0 +1,103 @@
+
+    @Mock
+    private DigiwfEmailApi digiwfEmailApi;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private RuntimeService runtimeService;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private ProcessConfigService processConfigService;
+
+    @Captor
+    private ArgumentCaptor<Mail> incidentEntityCaptor;
+
+    @InjectMocks
+    @Spy()
+    private IncidentNotifierHandlerMock incidentNotifierHandler;
+
+    /**
+     * Tests incident notifications for configured email addresses.
+     *
+     * @throws IOException        In case there is an issue with reading a file on the file system.
+     * @throws MessagingException In case there is an issue with sending an e-mail to the notification receivers.
+     */
+    @Test
+    void testIncidentNotifications() throws IOException, MessagingException {
+
+        // Setup
+        final IncidentContext incidentContext = mock(IncidentContext.class);
+        final IncidentEntity incidentEntity = mock(IncidentEntity.class);
+        when(incidentEntity.getProcessInstanceId()).thenReturn("");
+        when(runtimeService
+                .createProcessInstanceQuery()
+                .processInstanceId(anyString())
+                .singleResult()
+                .getRootProcessInstanceId())
+                .thenReturn("rootProcessInstanceId");
+        when(runtimeService
+                .createProcessInstanceQuery()
+                .processInstanceId(anyString())
+                .singleResult()
+                .getProcessDefinitionId())
+                .thenReturn(CONFIG_KEY + ":v42:12345");
+        ProcessConfig.builder()
+                .key("processDefinitionId");
+        final String configContent = Files.readString(Paths.get("src/test/resources/dummy/Test.processconfig.json"));
+        final ProcessConfigEntity processConfigEntity = ProcessConfigEntity.builder()
+                .version("v1")
+                .key(CONFIG_KEY)
+                .config(configContent)
+                .build();
+        final ProcessConfig processConfig = new ProcessConfigMapper().map(processConfigEntity);
+        when(processConfigService.getProcessConfig(anyString())).thenReturn(Optional.of(processConfig));
+        when(incidentNotifierHandler.superHandleIncident(any(), anyString())).thenReturn(incidentEntity);
+
+        // Execute
+        incidentNotifierHandler.handleIncident(incidentContext, "Incident message");
+
+        // Assert that an email was sent with the correct receivers.
+        verify(digiwfEmailApi, times(1)).sendMailWithDefaultLogo(incidentEntityCaptor.capture());
+        final Mail mail = incidentEntityCaptor.getValue();
+        assertThat(mail.getReceivers()).isEqualTo(processConfig.getIncidentNotificationAddresses());
+    }
+
+    private static class IncidentNotifierHandlerMock extends IncidentNotifierHandler {
+
+        // Workaround: If a static method is in the chain, Mockito will not mock method calls of superclasses.
```

Comment:
Könnte man das nicht wie nachfolgend Beschrieben lösen?
https://stackoverflow.com/questions/3467801/mockito-how-to-mock-only-the-call-of-a-method-of-the-superclass
Würde ich fast schöner finden einfach das `createIncident` Incident zu mocken und da null zurück zu geben

Reply:
Genau das hatte ich probiert, aber weil irgendwann IncidentEntity.create aufgerufen wird und das eine static-Methode ist, geht es leider nicht. Dafür müssten wir Power-Mockito einführen, was es in dem Projekt noch nicht gibt.